src/dotnet/Chat.UI.Blazor/Components/Onboarding/RoundProgress.razor

```diff
@@ -0,0 +1,40 @@
+@namespace ActualChat.Chat.UI.Blazor.Components
+
+<div class="round-progress">
+    <svg width="@Diameter" height="@Diameter">
+        <circle
+            class="text-icons-02"
+            stroke-width="1"
+            stroke="currentColor"
+            fill="transparent"
+            r="@Radius"
+            cx="@Center"
+            cy="@Center"/>
+        <circle
+            class="text-icons-04"
+            stroke-width="2"
+            stroke-dasharray="@Circumference"
+            stroke-dashoffset="@Offset"
+            transform="rotate(-90,@Center,@Center)"
+            stroke-linecap="round"
+            stroke="currentColor"
+            fill="transparent"
+            r="@Radius"
+            cx="@Center"
+            cy="@Center"/>
+    </svg>
+    @ChildContent
+</div>
+
+@code {
+    [Parameter] public int TotalSteps { get; set; }
+    [Parameter] public int CurrentStep { get; set; }
+    [Parameter] public double Diameter { get; set; } = 58;
+    [Parameter] public RenderFragment? ChildContent { get; set; }
+
+    private double Center => Diameter / 2;
+    private double Radius => Diameter / 2 - 2;
+    private double Circumference => 2 * Math.PI * Radius;
+    private double ProgressPercent => TotalSteps == 0 ? 0 : CurrentStep / ((double)TotalSteps / 100);
+    private double Offset => Circumference - ProgressPercent / 100 * Circumference;
```

Comment:
looks like it's better to store progress and do all the math with progress, while progressPercent is more like representual property